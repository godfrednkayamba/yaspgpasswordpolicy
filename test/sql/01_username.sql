LOAD 'yaspgpasswordpolicy';
--
--reset all settings
--
SET yaspgpp.username_min_length TO DEFAULT;
SET yaspgpp.username_min_special TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_digit TO DEFAULT;
SET yaspgpp.username_contain_password TO DEFAULT;
SET yaspgpp.username_ignore_case TO DEFAULT;
SET yaspgpp.username_contain TO DEFAULT;
SET yaspgpp.username_not_contain TO DEFAULT;
SET yaspgpp.username_min_repeat TO DEFAULT;
SET yaspgpp.password_min_length TO DEFAULT;
SET yaspgpp.password_min_special TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_digit TO DEFAULT;
SET yaspgpp.password_contain_username TO DEFAULT;
SET yaspgpp.password_ignore_case TO DEFAULT;
SET yaspgpp.password_contain TO DEFAULT;
SET yaspgpp.password_not_contain TO DEFAULT;
SET yaspgpp.password_min_repeat TO DEFAULT;

--username checks
--
--length must be >=2
--
SET yaspgpp.username_min_length TO 2;
DROP USER IF EXISTS a;
CREATE USER a WITH PASSWORD 'dummy';
DROP USER IF EXISTS a;
CREATE USER a;
DROP USER IF EXISTS a;
--
--min user repeat
--
SET yaspgpp.username_min_repeat TO 5;
DROP USER IF EXISTS abbbaaaaaa;
CREATE USER abbbaaaaaa WITH PASSWORD 'dummy';
DROP USER IF EXISTS abbbaaaaaa;
--
--min special >= 1
--
SET yaspgpp.username_min_special TO 1;
DROP USER IF EXISTS a$;
CREATE USER aa WITH PASSWORD 'dummy';
CREATE USER a$ WITH PASSWORD 'dummy';
DROP USER IF EXISTS a$;
--
--min upper >=1
--
SET yaspgpp.username_min_upper TO 1;
DROP USER IF EXISTS "aA$";
CREATE USER "aa$" WITH PASSWORD 'dummy';
CREATE USER "aA$" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aA$";
--
--min lower >=2
--
SET yaspgpp.username_min_lower TO 1;
DROP USER IF EXISTS "AAA$";
CREATE USER "AAA$" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$";
CREATE USER "aaA$" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$";
--
--must contain one of the characters 'a','b','c'
--
SET yaspgpp.username_contain TO 'a,b,c';
DROP USER IF EXISTS "pA$user";
CREATE USER "pA$user" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aA$user";
CREATE USER "aA$user" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aA$user";
--
--must not contain one of the characters 'x','z'
--
SET yaspgpp.username_not_contain TO 'x,z';
DROP USER IF EXISTS "xaA$user";
CREATE USER "xaA$user" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$user";
CREATE USER "aaA$user" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$user";
--
--username contain password
--
SET yaspgpp.username_contain_password TO on;
DROP USER IF EXISTS "aaA$dummy";
CREATE USER "aaA$dummy" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$usernopass";
CREATE USER "aaA$usernopass" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$usernopass";
--
--ignore case while performing checks
--
SET yaspgpp.username_ignore_case TO on;
DROP USER IF EXISTS "aa$user_dummy";
CREATE USER "aa$user_dummy" WITH PASSWORD 'DUMMY';
DROP USER IF EXISTS "aa$user_DUMMY";
CREATE USER "aa$user_DUMMY" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aa$user_dummy";
--
--min digit >=1
--
SET yaspgpp.username_min_digit TO 1;
DROP USER IF EXISTS aa;
CREATE USER aa WITH PASSWORD 'dummy';
DROP USER IF EXISTS aa2;
CREATE USER aa2 WITH PASSWORD 'dummy';
DROP USER IF EXISTS aa2;

CREATE USER "a$user1" WITH PASSWORD '';
DROP USER "a$user1";
CREATE USER aa;
DROP USER aa;

