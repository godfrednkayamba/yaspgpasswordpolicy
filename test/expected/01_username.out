LOAD 'yaspgpp';
--
--reset all settings
--
SET yaspgpp.username_min_length TO DEFAULT;
SET yaspgpp.username_min_special TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_digit TO DEFAULT;
SET yaspgpp.username_contain_password TO DEFAULT;
SET yaspgpp.username_ignore_case TO DEFAULT;
SET yaspgpp.username_contain TO DEFAULT;
SET yaspgpp.username_not_contain TO DEFAULT;
SET yaspgpp.username_min_repeat TO DEFAULT;
SET yaspgpp.password_min_length TO DEFAULT;
SET yaspgpp.password_min_special TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_digit TO DEFAULT;
SET yaspgpp.password_contain_username TO DEFAULT;
SET yaspgpp.password_ignore_case TO DEFAULT;
SET yaspgpp.password_contain TO DEFAULT;
SET yaspgpp.password_not_contain TO DEFAULT;
SET yaspgpp.password_min_repeat TO DEFAULT;
--username checks
--
--length must be >=2
--
SET yaspgpp.username_min_length TO 2;
DROP USER IF EXISTS a;
NOTICE:  role "a" does not exist, skipping
CREATE USER a WITH PASSWORD 'dummy';
ERROR:  username length should match the configured yaspgpp.username_min_length (2)
DROP USER IF EXISTS a;
NOTICE:  role "a" does not exist, skipping
CREATE USER a;
ERROR:  username length should match the configured yaspgpp.username_min_length (2)
DROP USER IF EXISTS a;
NOTICE:  role "a" does not exist, skipping
--
--min user repeat
--
SET yaspgpp.username_min_repeat TO 5;
DROP USER IF EXISTS abbbaaaaaa;
NOTICE:  role "abbbaaaaaa" does not exist, skipping
CREATE USER abbbaaaaaa WITH PASSWORD 'dummy';
ERROR:  username characters are repeated more than the configured yaspgpp.username_min_repeat times (5)
DROP USER IF EXISTS abbbaaaaaa;
NOTICE:  role "abbbaaaaaa" does not exist, skipping
--
--min special >= 1
--
SET yaspgpp.username_min_special TO 1;
DROP USER IF EXISTS a$;
NOTICE:  role "a$" does not exist, skipping
CREATE USER aa WITH PASSWORD 'dummy';
ERROR:  username does not contain the configured yaspgpp.username_min_special characters (1)
CREATE USER a$ WITH PASSWORD 'dummy';
DROP USER IF EXISTS a$;
--
--min upper >=1
--
SET yaspgpp.username_min_upper TO 1;
DROP USER IF EXISTS "aA$";
NOTICE:  role "aA$" does not exist, skipping
CREATE USER "aa$" WITH PASSWORD 'dummy';
ERROR:  username does not contain the configured yaspgpp.username_min_upper characters (1)
CREATE USER "aA$" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aA$";
--
--min lower >=2
--
SET yaspgpp.username_min_lower TO 1;
DROP USER IF EXISTS "AAA$";
NOTICE:  role "AAA$" does not exist, skipping
CREATE USER "AAA$" WITH PASSWORD 'dummy';
ERROR:  username does not contain the configured yaspgpp.username_min_lower characters (1)
DROP USER IF EXISTS "aaA$";
NOTICE:  role "aaA$" does not exist, skipping
CREATE USER "aaA$" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$";
--
--must contain one of the characters 'a','b','c'
--
SET yaspgpp.username_contain TO 'a,b,c';
DROP USER IF EXISTS "pA$user";
NOTICE:  role "pA$user" does not exist, skipping
CREATE USER "pA$user" WITH PASSWORD 'dummy';
ERROR:  username does not contain the configured yaspgpp.username_contain characters: a,b,c
DROP USER IF EXISTS "aA$user";
NOTICE:  role "aA$user" does not exist, skipping
CREATE USER "aA$user" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aA$user";
--
--must not contain one of the characters 'x','z'
--
SET yaspgpp.username_not_contain TO 'x,z';
DROP USER IF EXISTS "xaA$user";
NOTICE:  role "xaA$user" does not exist, skipping
CREATE USER "xaA$user" WITH PASSWORD 'dummy';
ERROR:  username contains the configured yaspgpp.username_not_contain unauthorized characters: x,z
DROP USER IF EXISTS "aaA$user";
NOTICE:  role "aaA$user" does not exist, skipping
CREATE USER "aaA$user" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$user";
--
--username contain password
--
SET yaspgpp.username_contain_password TO on;
DROP USER IF EXISTS "aaA$dummy";
NOTICE:  role "aaA$dummy" does not exist, skipping
CREATE USER "aaA$dummy" WITH PASSWORD 'dummy';
ERROR:  username should not contain password
DROP USER IF EXISTS "aaA$usernopass";
NOTICE:  role "aaA$usernopass" does not exist, skipping
CREATE USER "aaA$usernopass" WITH PASSWORD 'dummy';
DROP USER IF EXISTS "aaA$usernopass";
--
--ignore case while performing checks
--
SET yaspgpp.username_ignore_case TO on;
DROP USER IF EXISTS "aa$user_dummy";
NOTICE:  role "aa$user_dummy" does not exist, skipping
CREATE USER "aa$user_dummy" WITH PASSWORD 'DUMMY';
ERROR:  username should not contain password
DROP USER IF EXISTS "aa$user_DUMMY";
NOTICE:  role "aa$user_DUMMY" does not exist, skipping
CREATE USER "aa$user_DUMMY" WITH PASSWORD 'dummy';
ERROR:  username should not contain password
DROP USER IF EXISTS "aa$user_dummy";
NOTICE:  role "aa$user_dummy" does not exist, skipping
--
--min digit >=1
--
SET yaspgpp.username_min_digit TO 1;
DROP USER IF EXISTS aa;
NOTICE:  role "aa" does not exist, skipping
CREATE USER aa WITH PASSWORD 'dummy';
ERROR:  username does not contain the configured yaspgpp.username_min_digit characters (1)
DROP USER IF EXISTS aa2;
NOTICE:  role "aa2" does not exist, skipping
CREATE USER aa2 WITH PASSWORD 'dummy';
ERROR:  username does not contain the configured yaspgpp.username_min_special characters (1)
DROP USER IF EXISTS aa2;
NOTICE:  role "aa2" does not exist, skipping
CREATE USER "a$user1" WITH PASSWORD '';
ERROR:  password length should match the configured yaspgpp.password_min_length (1)
DROP USER "a$user1";
ERROR:  role "a$user1" does not exist
CREATE USER aa;
ERROR:  username does not contain the configured yaspgpp.username_min_digit characters (1)
DROP USER aa;
ERROR:  role "aa" does not exist
